### ISSUE TITLE
Add option to use psaw to (just/at least) get relevant IDs and praw to get current data

### ISSUE THREAD

**Issue by dmarx** [dmarx/psaw#10]
1. Add argument to `__init__` to pass a `praw.Reddit` object 
2. psaw downloads IDs and passes to praw using an [async queue](http://asyncio.readthedocs.io/en/latest/producer_consumer.html)
3. praw pulls data (asynchronously wrt psaw, probably using [coroutines?](https://docs.python.org/3/library/asyncio-task.html))

Might be cool for psaw to (optionally) get data from pushshift in addition to IDs in case content has been removed/deleted, or alternatively have an argument for psaw to collect IDs for deleted/removed objects and then go back to pushshift to get their data. Probably way slower this way than just getting the data the first time and throwing it away after confirming we don't need it. Call this a low priority, but give users the option to do things this way. 

Make default behavior getting all the relevant data from pushshift and updating from praw where needed.

**Comment by dmarx** [comment]
Getting reddit "things" by ID in batches (of 100):

https://praw.readthedocs.io/en/latest/code_overview/reddit_instance.html#praw.Reddit.info

* subm prefix: t3
* comm prefix: t1

**Comment by dmarx** [comment]
see dev  80651b1

demo:

```
import praw
from psaw import PushshiftAPI #, WithPraw

r = praw.Reddit(...)

api = PushshiftAPI()

for c in api.search_comments(author='someredditor', limit=1):
    print(type(c), c.score) # <class 'psaw.PushshiftAPI.comment'> 1

api = PushshiftAPI(r)
    
for c in api.search_comments(author='someredditor', limit=1):
    print(type(c), c.score) # <class 'praw.models.reddit.comment.Comment'> 134

```

**Comment by dmarx** [comment]
Published in v0.0.3
